x-pack/plugins/reporting/public/share_context_menu/register_pdf_png_reporting.tsx

```diff
@@ -93,7 +94,8 @@ export const reportingScreenshotShareProvider = ({
       capabilityHasVisualizeScreenshotReporting = true;
     }
 
-    if (!licenseHasScreenshotReporting) {
+    // do not want screenshot reporting showing for serverless
+    if (!licenseHasScreenshotReporting || !enable) {
```

Comment:
Based on my suggestion above, if sharing is not enabled from this type, we should not register the provider in plugin.ts. That should simplify the changes we need to make in this PR a bit.